scripts/hydrogen_bond_propensity/hydrogen_bond_propensity_report.py

```diff
@@ -64,6 +66,56 @@ def make_diagram(mol, directory):
     return fname
 
 
+def make_diagram_components(crystal, molecule, directory, docx_template):
+    # Generates a diagram for each component in a given structure, with D/A atoms labelled if component has a donor/acceptor
+    molecule_diagram_generator = DiagramGenerator()
+    molecule_diagram_generator.settings.line_width = 1.6
+    molecule_diagram_generator.settings.font_size = 10
+    molecule_diagram_generator.settings.image_height = 300
+    molecule_diagram_generator.settings.shrink_symbols = False
+
+    asymmetric_molecule = crystal.asymmetric_unit_molecule
+    non_ch_molecule_no = 0
+    component_diagrams = []
+    donor_acceptor_check = []
+
+    # If no Z and Z' information (e.g. mol2), 'None' is returned for crystal.z_prime
+    z_prime_flag = False
+    if crystal.z_prime:
+        z_prime_flag = True
+
+    if z_prime_flag and crystal.z_prime < 1:
+        for m in molecule.components:  # generates an image for every molecule
+            atoms = m.atoms
+            donor_acceptor_check = [atom for atom in atoms if atom.is_donor or atom.is_acceptor]
+
+            if donor_acceptor_check:
+                lbl_da_atoms = [atom for atom in atoms if atom.is_donor or atom.is_acceptor]
```

Comment:
This can re-use the above `donor_acceptor_check`

Reply:
Yep, will switch to that